core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -516,16 +516,22 @@ class ZkPartitionStateMachine(config: KafkaConfig,
 }
 
 object PartitionLeaderElectionAlgorithms {
-  def offlinePartitionLeaderElection(assignment: Seq[Int], isr: Seq[Int], liveReplicas: Set[Int], uncleanLeaderElectionEnabled: Boolean, controllerContext: ControllerContext): Option[Int] = {
-    assignment.find(id => liveReplicas.contains(id) && isr.contains(id)).orElse {
-      if (uncleanLeaderElectionEnabled) {
-        val leaderOpt = assignment.find(liveReplicas.contains)
-        if (leaderOpt.isDefined)
-          controllerContext.stats.uncleanLeaderElectionRate.mark()
-        leaderOpt
-      } else {
-        None
-      }
+
+  /**
+   * @return Optionally, a tuple (replica, flag) where flag is a boolean indicating if unclean leader election was
+   *         used to replace the replica.
+   */
+  def offlinePartitionLeaderElection(assignment: Seq[Int], isr: Seq[Int], liveReplicas: Set[Int], uncleanLeaderElectionEnabled: Boolean): Option[(Int, Boolean)] = {
+    assignment.find(id => liveReplicas.contains(id) && isr.contains(id)) match {
+      case Some(replicaId) => Some(replicaId, false)
+      case None =>  if (uncleanLeaderElectionEnabled) {
+                     assignment.find(liveReplicas.contains) match {
+                        case Some(uncleanReplicaId) => Some(uncleanReplicaId, true)
+                        case None => None
+                      }
+                  } else {
+                    None
+                  }
```

Comment:
This formatting doesn't seem correct. How about:
```scala
assignment
  .find(id => liveReplicas.contains(id) && isr.contains(id))
  .map((_, false)) // or .map(_ -> false) or .map(id => (id, false))
  .orElse {
    if (uncleanLeaderElectionEnabled) {
      assignment.find(liveReplicas.contains).map((_, true))
    else {
      None
    }
  }
```

Reply:
This won't compile. At least on scala 2.11.  The compiler thinks the type for the first type parameter of the tuple is Any or something else.